src/crimson/osd/pg.cc

```diff
@@ -838,13 +838,18 @@ PG::do_osd_ops(
 {
   auto ox = std::make_unique<OpsExecuter>(
     std::move(obc), op_info, get_pool().info, get_backend(), msg_params);
-  return do_osd_ops_execute<void>(
-    std::move(*ox),
-    std::move(ops),
-    std::as_const(op_info),
-    std::move(success_func),
-    std::move(failure_func)
-  ).finally([ox_deleter=std::move(ox)] {});
+  return seastar::do_with(std::move(ops), 
```

Comment:
per our discussion offline, might want to move the `do_with()` out of this method, so these two variants of `PG::do_osd_ops()` can be more consistent from this perspective: both of them rely on the caller to ensure that `ops` is alive before the future of the callee continuation is fulfilled.